cli/src/commands/init.rs

```diff
@@ -120,35 +120,42 @@ fn prompt_group() -> io::Result<Option<String>> {
 fn prompt_lockfile(
     cli_lockfile: Option<&String>,
     cli_lockfile_type: Option<&String>,
-) -> io::Result<(String, String)> {
-    let lockfile = match (cli_lockfile.cloned(), cli_lockfile_type) {
+) -> io::Result<Vec<LockfileConfig>> {
+    let mut lockfiles = match (cli_lockfile.cloned(), cli_lockfile_type) {
         // Do not prompt if name and type were specified on CLI.
-        (Some(lockfile), Some(lockfile_type)) => return Ok((lockfile, lockfile_type.clone())),
+        (Some(lockfile), Some(lockfile_type)) => {
+            let lockfiles = vec![LockfileConfig::new(lockfile, lockfile_type.into())];
+            return Ok(lockfiles);
+        },
         // Prompt for type if only lockfile was passed.
-        (Some(lockfile), _) => lockfile,
+        (Some(lockfile), _) => vec![lockfile],
         // Prompt for lockfile if it wasn't specified.
         (None, _) => prompt_lockfile_name()?,
     };
 
-    // Try to determine lockfile name from known lockfiles.
-    for format in LockfileFormat::iter() {
-        if format.parser().is_path_lockfile(Path::new(&lockfile)) {
+    // Find lockfile type for each lockfile.
+    let mut lockfile_configs = Vec::new();
+    for lockfile in lockfiles.drain(..) {
+        // Try to determine lockfile type from known formats.
+        if let Some(format) = LockfileFormat::iter()
+            .find(|format| format.parser().is_path_lockfile(Path::new(&lockfile)))
+        {
             let lockfile_type = format.name().to_owned();
-            return Ok((lockfile, lockfile_type));
+            lockfile_configs.push(LockfileConfig::new(lockfile, lockfile_type));
+            continue;
         }
-    }
 
-    // Prompt for lockfile type.
-    let lockfile_type = match cli_lockfile_type {
-        Some(lockfile_type) => lockfile_type.clone(),
-        None => prompt_lockfile_type()?,
-    };
+        // Prompt for lockfile type.
+        let lockfile_type = prompt_lockfile_type(&lockfile)?;
 
-    Ok((lockfile, lockfile_type))
+        lockfile_configs.push(LockfileConfig::new(lockfile, lockfile_type));
+    }
+
+    Ok(lockfile_configs)
 }
 
 /// Ask for the lockfile name.
-fn prompt_lockfile_name() -> io::Result<String> {
+fn prompt_lockfile_name() -> io::Result<Vec<String>> {
```

Comment:
It is not evident that `prompt_lockfile_name` potentially returns multiple lockfiles; `prompt_lockfile_names` would probably be better here.